packages/reader-syntax-trees-mode/src/SyntaxTreesModeReader.vue

```diff
@@ -0,0 +1,176 @@
+<template>
+  <ApolloQuery
+    class="reader"
+    :query="query"
+    :variables="queryVariables"
+    :update="queryUpdate"
+  >
+    <template v-slot="{ result: { error, data }, isLoading }">
+      <LoaderBall v-if="isLoading" />
+      <ErrorMessage v-else-if="error">
+        There was an error loading the requested data.
+      </ErrorMessage>
+      <EmptyMessage v-else-if="!data" />
+      <template v-else>
+        <div class="sentence">
+          <span v-for="word in data.words" :key="word.id" :class="{ selected: selected(word) }" class="word" @mouseenter="onWordEnter(word)" @mouseleave="onWordLeave(word)">
+            {{ word.value }}
+          </span>
+        </div>
+        <Treant
+          class="syntax-tree"
+          :tree="data.tree"
+          :highlightedNode="hoveringOn"
+          @enter="onEnter"
+          @leave="onLeave"
+        />
+      </template>
+    </template>
+  </ApolloQuery>
+</template>
+
+<script>
+  import gql from 'graphql-tag';
+  import { ApolloQuery } from 'vue-apollo';
+
+  import { LoaderBall, ErrorMessage, EmptyMessage } from '@scaife-viewer/common';
+  import Treant from '@scaife-viewer/vue-treant';
+
+  const transformForTreant = node => {
+    const text = node.value === null
+      ? { name: 'Root' }
+      : { name: node.relation, desc: node.value, id: node.id};
+
+    return {
+      text,
+      children: node.children.map(child => transformForTreant(child)),
+    };
+  };
+
+  export default {
+    readerConfig: {
+      label: 'Syntax Trees',
+      textWidth: 'wide',
+    },
+    components: { ApolloQuery, LoaderBall, ErrorMessage, EmptyMessage, Treant },
+    props: {
+      queryVariables: Object
+    },
+    data() {
+      return {
+        hoveringOn: null,
+      };
+    },
+    methods: {
+      onWordEnter({ id }) {
+        this.hoveringOn = id;
+      },
+      onWordLeave(word) {
+        this.hoveringOn = null;
+      },
+      onEnter({ id }) {
+        this.hoveringOn = id;
+      },
+      onLeave() {
+        this.hoveringOn = null;
+      },
+      selected(word) {
+        return word.id === this.hoveringOn;
+      },
+      queryUpdate(data) {
+        const words = data.textAnnotations.edges
+          .map(e => e.node.data.words.map(word => {
+            return {
+              ...word,
+              children: [],
+            }
+          })).flat();
+        const wordBank = words.reduce((map, word) => {
+          return {
+            ...map,
+            [word.id]: word,
+          };
+        }, {});
+        const tree = {id: 0, value: null, headId: null, relation: null, children: []};
+
+        words.forEach(word => {
+          if (word.headId === 0) {
+            tree.children.push(word);
+          } else {
+            wordBank[word.headId].children.push(word);
+          }
+        });
+        return {
```

Comment:
two comments here:

1) If you paginate to the next passage, you get an empty "Root":

![image](https://user-images.githubusercontent.com/629062/92811767-5d9ca100-f384-11ea-8459-06e5341ce920.png)


I think that we need to check  in`<EmptyMessage v-else-if="!data" />` off of something other than data that has a falsey value..likely checking the number of nodes on the edge, for example

2) related to the above, this structure assumes that there is only _one_ syntax tree for a given passage.  I will expand the dataset and ping back here when deployed so we can have a data stub with multiple trees in a single passage.

Reply:
@paltman I've loaded more data into ATLAS.

If you paginate from 1.7 to 1.8, there seems to be a reactivity issue:

https://deploy-preview-21--scaife-viewer-frontend.netlify.app/explore-homer?urn=urn%3Acts%3AgreekLit%3Atlg0012.tlg001.perseus-grc2%3A1.7

The sentence / words changes, but the graph does not:

https://share.getcloudapp.com/04uqZb1W

Reply:
`1.9` should have two trees:

https://deploy-preview-21--scaife-viewer-frontend.netlify.app/explore-homer?urn=urn%3Acts%3AgreekLit%3Atlg0012.tlg001.perseus-grc2%3A1.9

https://www.perseids.org/tools/arethusa/app/#/perseids?chunk=3&doc=28550

https://www.perseids.org/tools/arethusa/app/#/perseids?chunk=4&doc=28550

Reply:
How should we show multiple trees?

Reply:
@paltman If you look at the arethusa links above, they are paginating there by "chunk" (that is, sentences).

When we groomed the syntax trees (and developed the ATLAS query), we're returning sentence(s) that map to the current passage, so there could be more than one.

This is similar to what we're doing with the metrical display mode or the "old" sentence alignments:

_1.1-1.5: 1 alignment_

<img width="1238" alt="image" src="https://user-images.githubusercontent.com/629062/92965658-c98e1080-f43b-11ea-91ff-e690297f1d51.png">
<br />

_1.6-1.10: 4 alignments_

<img width="1238" alt="image" src="https://user-images.githubusercontent.com/629062/92965682-d3177880-f43b-11ea-876a-fc5aa7a2cb3a.png">
<br />

_1.11-1.15: 2 alignments_

<img width="1238" alt="image" src="https://user-images.githubusercontent.com/629062/92965737-ede9ed00-f43b-11ea-81bc-0ac53a98c2f2.png">
<br />


(as an aside, folio image mode _should_ work like the sentence alignments, but is rendering just the first folio image; I'll capture an issue for further elaboration on that.)

I can think of a couple of way to handle multiple sentences for a single passage:

1) Short term, render the first tree, and provide an indication in the UX that additional _n_ trees are not being shown.  The `words` and `tree` data structures you're building would need to _only_ parse data from the first node returned if there are multiple trees.

2) Ideally, we're rendering multiple sentences within the reader mode.  You build up a list of `words` and `tree` data structures, and you loop through "rows" (just like the metrical mode).

```
# # # # # # # # # # # # # # # # # # # # # # # #
# sentence 1
# graph 1
# # # # # # # # # # # # # # # # # # # # # # # #

# # # # # # # # # # # # # # # # # # # # # # # #
# sentence2
# graph2
# # # # # # # # # # # # # # # # # # # # # # # #
```

If you toggled only sentences, you'd only see sentences, only graphs, only the graphs etc.

I'd suggested on our grooming call that we might want an accordion like toggle on the either the "row" or maybe just the "graph" so you could quickly collapse / expand the graph (e.g., the container, not the "nodes" in the graph):

_Figure A: expand/collapse entire rows_

```
# # # # # # # # # # # # # # # # # # # # # # # #
# [expanded \/ ]
# sentence1
# graph1
# # # # # # # # # # # # # # # # # # # # # # # #

# # # # # # # # # # # # # # # # # # # # # # # #
# [collapsed  > ]
# sentence1
# graph1
# # # # # # # # # # # # # # # # # # # # # # # #
```

_Figure B: expand/collapse only graphs_

```
# # # # # # # # # # # # # # # # # # # # # # # #
# sentence1
# graph1 [expanded \/ ]
# # # # # # # # # # # # # # # # # # # # # # # #

# # # # # # # # # # # # # # # # # # # # # # # #
# sentence2
# graph2 [collapsed > ]
# # # # # # # # # # # # # # # # # # # # # # # #
```

I think _Figure B_ would be better UX

Reply:
I agree, Figure B would work better. Then you can keep the context of the text while viewing just a single tree.